addons/controllers/addon_controller.go

```diff
@@ -423,10 +427,6 @@ func (r *AddonReconciler) reconcileAddonSecretNormal(
 
 	*patchAddonSecret = metadataAdded
 
-	if r.shouldNotReconcile(log, addonSecret) {
```

Comment:
IMO This should be moved above the code that adds finalizer. We should not add finalizer to the addon secret if it's paused.